internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -787,6 +787,7 @@ func expandEncryption(d *pluginsdk.ResourceData) *vaults.VaultPropertiesEncrypti
 	}
 	if v, ok := encryptionMap["user_assigned_identity_id"].(string); ok && v != "" {
 		encryption.KekIdentity.UserAssignedIdentity = utils.String(v)
+		encryption.KekIdentity.UseSystemAssignedIdentity = utils.Bool(false)
```

Comment:
Instead of silently changing this value behind the scenes maybe instead we can error out while returning a more descriptive error that tells users that they must set the other property to false?

Reply:
Hi @katbyte,

Thanks for this suggestion. I made changes to throw the error now.